doc/nyan/base types.md

```diff
@@ -0,0 +1,164 @@
+Base types
+==========
+
+These types are defined by the openage engine. Every nyan object in the game
+eventually inherits from one or more of these types.
+
+Normally you'd store one or more *NyanObject*s in a *.nyan file, but as these
+are defined in the engine and are here only for humans (like me, *beep blop*),
+they are stored here for an overview.
+
+Currently, this is a **proposal**.
+This file does not perfectly (if at all) represent the already defined types in
+libopenage, but hopefully will in the future. Many classes here, if not all,
+are currently work in progress. Prepare for changes.
+
+Shoutout:
+* Everyone: Comment! Give us good ideas.
+* libopenage-devs: change/erase this. You are the ones, who (are going to)
+implement this and know what's necessary and what should be where.
+
+```
+Mod():
+	# All patches in the 'patches' member are added on load time. So every mod
+	# needs to inherit from this class and new patches to 'patches'.
+
+	name : text
+	patches : set(NyanPatch) # Add your mod's patches to this member.
```

Comment:
Yes, that is the idea from the nyan doc currently, but we've come up with an alternative (which i still have to document):

If a mod is defined by instancing a child object of `Mod`, we have to support callback functions for objects if a child of them is created. This is doable, but kinda complicated.

New idea:
All the data (name, dependencies, etc) is defined as part of the mod metainformation, and the set of patches as well. Because then we don't have to implement that child hook and the initial patches can simply be applied by the engine directly. See #632